ipfs/defaults.go

```diff
@@ -0,0 +1,201 @@
+		Swarm: ipfscfg.SwarmConfig{
+			// auto port forwarding is nice
+			DisableNatPortMap: false,
+			// metrics are useful
+			DisableBandwidthMetrics: false,
+			// relays are cool from engineering standpoint, but don't use them
+			EnableRelayHop: false, EnableAutoRelay: false,
+			// ConnMgr manager controls the amount of conns a Node holds. If the amount of conns hits HighWater, ConnMgr
+			// trims amount of connections down to LowWater
+			// NOTE: Connections can by prioritized using Tagging in ConnMgr in order to keep important conns alive
+			ConnMgr: ipfscfg.ConnMgr{
+				// we use defaults from IPFS for HighWater
+				HighWater: 900,
+				// we use defaults from IPFS for HighWater
+				LowWater: 600,
+				// defines how often ConnMgr should check conns for trimming, we use default from IPFS
+				GracePeriod: "20s",
+				// just a basic connection manager
+				Type: "basic",
+			},
+			// Configuration for all networking protocols
+			Transports: ipfscfg.Transports{
+				Network: transport{
+					// we default to TCP as it's most common, battle tested transport, but we may migrate to the one below
+					TCP: ipfscfg.False,
+					// we might default to QUIC at some point, but for now TCP is more common and reliable
+					QUIC: ipfscfg.False,
+					// we don't use Websocket as we don't target for browser nodes
+					Websocket: ipfscfg.False,
+					// we don't use relays as it's expensive in terms of traffic for relay nodes
+					Relay: ipfscfg.False,
+				},
+				// Security protocol wraps IO conns in a private encrypted tunnel
+				Security: security{
+					// we default to Noise, as it's designed for p2p interactions in the first place and is 0-RTT
+					Noise: ipfscfg.DefaultPriority,
+					// we don't use SECIO as it's now deprecated and has know issues
+					SECIO: ipfscfg.Disabled,
+					// we don't allow TLS option, even it's a production ready protocol, to lower bug vectors
+					TLS: ipfscfg.Disabled,
+				},
+				// Multiplexers provide ability to run multiple logical streams over one transport stream, e.g TCP conn
+				Multiplexers: multiplexers{
+					// we default to Yamux due to production readiness
+					Yamux: ipfscfg.DefaultPriority,
+					// we don't use Mplex as it's more like a toy multiplexer for testings and etc
+					Mplex: ipfscfg.Disabled,
+				},
+			},
+			// this ia manual peer blacklist
+			AddrFilters: nil,
+		},
+		// Persistent KV store configuration
+		Datastore: ipfscfg.Datastore{
+			// Configuration for badger kv we default to
+			Spec: map[string]interface{}{
+				"type":   "measure",
+				"prefix": "badger.datastore",
+				"child": map[string]interface{}{
+					"type":       "badgerds",
+					"path":       "badgerds",
+					"syncWrites": false,
+					"truncate":   true,
+				},
+			},
+			// we don't use bloom filtered blockstore
+			BloomFilterSize: 0,
+			// we don't use GC so values below does not have any affect
+			StorageGCWatermark: 0,
+			GCPeriod:           "",
+			StorageMax:         "",
```

Comment:
Are those badger specific? Because garbage collection could mean different things in different contexts.

Reply:
Nope that's specific to IPFS GC